corporate/views/billing_page.py

```diff
@@ -288,20 +285,18 @@ def update_plan_for_remote_realm(
 
 @authenticated_remote_server_management_endpoint
 @process_as_post
-@has_request_variables
+@typed_endpoint
 def update_plan_for_remote_server(
     request: HttpRequest,
     billing_session: RemoteServerBillingSession,
-    status: Optional[int] = REQ(
-        "status",
-        json_validator=check_int_in(ALLOWED_PLANS_API_STATUS_VALUES),
-        default=None,
-    ),
-    licenses: Optional[int] = REQ("licenses", json_validator=check_int, default=None),
-    licenses_at_next_renewal: Optional[int] = REQ(
-        "licenses_at_next_renewal", json_validator=check_int, default=None
-    ),
-    schedule: Optional[int] = REQ("schedule", json_validator=check_int, default=None),
+    *,
+    status: Annotated[
+        Optional[Json[int]],
+        AfterValidator(lambda x: check_int_in(x, ALLOWED_PLANS_API_STATUS_VALUES)),
+    ] = None,
```

Comment:
Here `Annotated` should not be wrapping `Optional` as this would mean that the validator would try to validate `None`, which will fail since `check_int_in` is expecting a int. 
Instead it should be `Optional[Annotated[Json[int], AfterValidator .....`.
Do this for all the similar params.

Reply:
Yes, I missed that.

Reply:
Also is there a difference between using the validator `Literal` like the example below?

```python
ALLOWED_PLANS_API_STATUS_VALUES = Literal[
.... allowed int values here
]
Optional[Annotated[Json[ALLOWED_PLANS_API_STATUS_VALUES]]
```

Reply:
The main annoyance with that is that we need to define the list of allowed values twice; once as a list of integers, and again as a list of `Literal`s of integers.